windows/ReactTestApp/Manifest.h

```diff
@@ -0,0 +1,52 @@
+#pragma once
+
+#include <list>
+#include <map>
+#include <optional>
+#include <string>
+
+#include <nlohmann/json.hpp>
+
+using json = nlohmann::json;
+using namespace std;
+
+namespace winrt::ReactTestApp::implementation {
+    struct Component {
+        string appKey;
+        std::optional<string> displayName;
+        std::optional<map<string, string>> initialProperties;
+    };
+
+    struct Manifest {
+        string name;
+        string displayName;
+        list<Component> components;
+    };
+
+    template <typename T>
+    std::optional<T> get_optional(const json &j, const string &key)
+    try {
+        return j.at(key).get<T>();
+    }
+    catch (const json::exception &) {
```

Comment:
Does nlohmann/json have APIs that are exception free? We should use those if available.

Reply:
Changed it now. Not sure if it's the best possible solution. but probably better than catching exception

Reply:
It looks like `at()` may be using `find()` underneath as well so this is fine.